app/ante/evm/06_account_verification.go

```diff
@@ -39,7 +40,9 @@ func VerifyAccountBalance(
 		account = statedb.NewEmptyAccount()
 	}
 
-	if err := keeper.CheckSenderBalance(sdkmath.NewIntFromBigInt(account.Balance), txData); err != nil {
+	// Converts the balance to 18 decimals
+	adjustedBalance := utils.ConvertTo18Decimals(*account.Balance)
```

Comment:
if the decimal is already 18, this will add 12 more which means that the total decimal places will be 30. Please fix this case